### PR TITLE
Clear existing values in a proto before serialization

### PR DESCRIPTION
**Pull request by yanavlasov**
Commit Message:
Newer version of protobuf changed Protobuf::util::JsonStringToMessage semantics to "merge". This affects some tests that serialize different yaml string into the same message.

This change clears the message before serialization, ensuring that no existing values are "inherited" during serialization.

This change is a no-op with current protobuf version.

Risk Level: Medium.
Testing: Unit Tests
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

Signed-off-by: Yan Avlasov <yavlasov@google.com>

### CODE REVIEW

File: source/common/protobuf/utility.cc [hunk 1]
```diff
@@ -264,6 +264,8 @@ Protobuf::util::Status MessageUtil::loadFromJsonNoThrow(const std::string& json,
   // Let's first try and get a clean parse when checking for unknown fields;
   // this should be the common case.
   options.ignore_unknown_fields = false;
+  // Clear existing values (if any) from the destination message before serialization.
+  message.Clear();
```
Comment (htuch) [inline source/common/protobuf/utility.cc:268]: Makes sense, can you add a test that capture the regression?

**Review by htuch — APPROVED** [review]
LGTM, but needs a merge for conflicts.

**Comment by RyanTheOptimist** [comment]
@htuch friendly maintainer on-call ping

**Review by htuch — APPROVED** [review]

**Pull request merged**